packages/smooth_app/lib/pages/scan/scan_page_helper.dart

```diff
@@ -28,44 +27,56 @@ Future<void> openPersonalizedRankingPage(BuildContext context) async {
   await model.refresh();
 }
 
-Widget buildButtonsRow(BuildContext context, ContinuousScanModel model) {
-  final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
-  final ButtonStyle buttonStyle = ButtonStyle(
-    shape: MaterialStateProperty.all<RoundedRectangleBorder>(
-      RoundedRectangleBorder(
-        borderRadius: BorderRadius.circular(18.0),
-      ),
-    ),
-  );
-  return AnimatedOpacity(
-    opacity: areButtonsRendered(model) ? 0.8 : 0.0,
-    duration: const Duration(milliseconds: 50),
-    child: Padding(
-      padding: const EdgeInsets.symmetric(
-        vertical: VERY_SMALL_SPACE,
-        horizontal: MEDIUM_SPACE,
+class ButtonsRow extends StatelessWidget {
```

Comment:
As a Java boy, I would suggest a similarity in the name of the file and its main class, which is not remotely the case here: a new dedicated file would make sense.
In addition to that, please try to give a distinct name for classes: `ButtonsRow` is a bit vague, isn't it? Something like `ScanMatchHeaderRow` or `ScanHeader`?